### ISSUE TITLE
Migrate utility adds YAML front matter to non-markdown files

### ISSUE THREAD

**Issue by cmpadden** [facebook/docusaurus#6145]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [ ] I have tried the `npm run clear` or `yarn clear` command.
- [ ] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [ ] I have tried creating a repro with https://new.docusaurus.io.
- [ ] I have read the console error message carefully (if applicable).

### Description

When running the migration utility:

```
npx @docusaurus/migrate migrate  ./v1-website ./v2-website
```

An empty front-matter section is added to non-markdown files in the `docs/` directory. This includes and is not limited to files with the extensions: `.json`, `.svg`, `.drawio`, `.sh`, `.js`, `.json`, and `.yaml`.

It was also added to the `.gitignore` file in the `docs/` directory.

This results in an empty front-matter at the top of these files:

```
---
---
```

### Steps to reproduce

1. Create a `.gitignore` file in your `docs/` directory.
2. Run `npx @docusaurus/migrate migrate  ./v1-website ./v2-website`

### Expected behavior

Front-matter should only be added to documentation files like those with the `.md` and `.mdx` extension (and other supported file formats).

### Actual behavior

Front-matter was added to all files, eg. `.gitignore` became:

```
---
---
<my_ignored_files>
```

### Your environment

The latest version of `@docusaurus/migrate` was used as of this writing: `2.0.0-beta.13`

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by Josh-Cena** [comment]
Thanks for reporting! Will be fixed by #6146. After that, you can test again with `npx @docusaurus/migrate@canary`

**Comment by cmpadden** [comment]
> Thanks for reporting! Will be fixed by #6146. After that, you can test again with `npx @docusaurus/migrate@canary`

Great - thank you!
